staging/src/k8s.io/client-go/applyconfigurations/core/v1/pod.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by applyconfiguration-gen. DO NOT EDIT.
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+	v1 "k8s.io/client-go/applyconfigurations/meta/v1"
+)
+
+// PodApplyConfiguration represents an declarative configuration of the Pod type for use
+// with apply.
+type PodApplyConfiguration struct {
+	v1.TypeMetaApplyConfiguration   `json:",inline"`
+	v1.ObjectMetaApplyConfiguration `json:"metadata,omitempty"`
+	Spec                            *PodSpecApplyConfiguration   `json:"spec,omitempty"`
+	Status                          *PodStatusApplyConfiguration `json:"status,omitempty"`
+}
+
+// PodApplyConfiguration constructs an declarative configuration of the Pod type for use with
+// apply.
+func Pod(name, namespace string) *PodApplyConfiguration {
+	b := &PodApplyConfiguration{}
+	b.SetName(name)
+	b.SetNamespace(namespace)
+	b.SetKind("Pod")
+	b.SetAPIVersion("v1")
+	return b
+}
+
+// SetKind sets the Kind field in the declarative configuration to the given value.
+func (b *PodApplyConfiguration) SetKind(value string) *PodApplyConfiguration {
```

Comment:
You're generating `SetKind` and `SetAPIVersion` both for `Pod` and `v1.TypeMetaApplyConfiguration`. Can we skip the variables that have been embedded?

Reply:
@jpbetz I know we mentioned it would be done as a follow up, but if we directly embed TypeMeta and ObjectMeta in this PR, it should save us a decent amount of redundant generated code. Is that something we can consider doing?

Reply:
We're limited by the fact that we're generating setters that we want to be able to chain. E.g.:

```go
Deployment(name, ns).SetLabels(...).SetSpec(...)
```

But when`ObjectMeta` is embedded in `Deployment`, the `SetLabel()` function that get's added returns `TypeMeta`, so the calls don't chain. This is why I'm generating a `SetLabel()` function that returns `Deployment`